src/joy.c

```diff
@@ -285,27 +319,10 @@ uint8_t Joy_GetStickData(int nStJoyId)
 	}
 	else if (ConfigureParams.Joysticks.Joy[nStJoyId].nJoystickMode == JOYSTICK_REALSTICK)
 	{
-		int nSdlJoyId;
-		int nAxes;	/* How many axes are there on the corresponding SDL joystick? */
-
-		nSdlJoyId = ConfigureParams.Joysticks.Joy[nStJoyId].nJoyId;
-		if (nSdlJoyId < 0 || !bJoystickWorking[nSdlJoyId])
+		/* map to SDL stick and Axes */
+		int nSdlJoyId = Joy_ReadAxisConfig(nStJoyId, &JoyReading);
+		if (nSdlJoyId < 0 || !bJoystickWorking[nSdlJoyId]) {
```

Comment:
I think the "|| !bJoystickWorking[nSdlJoyId]" part here is redundant since the check is already done in the Joy_ReadAxisConfig() function?
Also. please put the curly brace on a new line to adhere the coding style, thanks!